### DISCUSSION TITLE
Tensor-Network Quantum Circuit Simulation/PastaQ as Backend Simulator/Extension

### DISCUSSION THREAD

**Discussion by Fe-r-oz** [GTorlai/PastaQ.jl#312] [Q&A] [answered]
Hi, @GTorlai!
 
Is it correct to infer that `PastaQ.jl` is more suited for implementing tensor-network circuit simulations than` ITensor.jl?`

`PastaQ.jl` has even dedicated page on circuit simulation documentation: https://gtorlai.github.io/PastaQ.jl/stable/Circuits.html
There was no mention of circuit simulation at` ITensors.jl `documentation (is it because it is somehow assumed that we can do circuit simulation or it's better to use a package such as `PastaQ.jl` that is built on top of `ITensor.jl`?)

Please have a brief look at [MPS quantum circuit simulation with non-local range gates](https://itensor.discourse.group/t/mps-quantum-circuit-simulation-with-non-local-long-range-gates-best-strategies/733,) as well.

I am working on this issue: https://github.com/amazon-braket/Braket.jl/issues/81. The task is "Ideally the simulator should be able to consume` Braket.jl `circuits and translate them automatically".  It seems to be that` PastaQ.jl `would be more suited for this task.

Kindly please provide guidance as there is no documentation about how to incorporate `PastaQ.jl` as backend/extension with other packages.

Thanks,

Best Regards,

**Comment by mtfishman** [comment, marked as answer]
The circuit simulation code in PastaQ.jl is just a wrapper around the one in ITensorMPS.jl, see for example https://itensor.github.io/ITensors.jl/dev/tutorials/MPSTimeEvolution.html and https://itensor.github.io/ITensors.jl/dev/MPSandMPO.html#Gate-evolution. It is just meant to be a little bit friendlier to users coming from quantum information, however I would recommend using ITensorMPS.jl now since PastaQ.jl has had limited development recently due to not having enough developers.

**Reply by Fe-r-oz** [reply]
Thanks for the update! After looking into `src/circuits/runcircuit.jl `, it makes better sense. 

https://github.com/GTorlai/PastaQ.jl/blob/d1a9116d01529eda7e47f1b17b7aa0ae68d04a3b/src/circuits/runcircuit.jl#L153-L160

https://github.com/GTorlai/PastaQ.jl/blob/d1a9116d01529eda7e47f1b17b7aa0ae68d04a3b/src/circuits/runcircuit.jl#L15-L29

I look forward to future developments!  I hope it finds more developers in the near future.

Best Regards,
